apps/tasks/lib/components/task_list_tile.dart

```diff
@@ -21,79 +21,101 @@ class TaskDTO{
     required this.subTitle,
     this.isChecked = false,
   });
-
 }
 
 /// A ListTile to Display a StaticProgressIndicator and Status
-class TaskListTile extends StatefulWidget{
-
+class TaskListTile extends StatefulWidget {
   /// Task data
   final TaskDTO task;
 
   /// Index for [ReorderableDragStartListener]
   final int index;
 
+  /// onDismissed Callback for [Dismissible]
+  final Function(DismissDirection?) onDismissed;
+
+  /// confirmDismiss Callback for [Dismissible]
+  final Future<bool> Function(DismissDirection direction) confirmDismiss;
+
   const TaskListTile({
     super.key,
     required this.task,
-    required this.index
+    required this.index,
+    required this.onDismissed,
+    required this.confirmDismiss
   });
 
 
   @override
   State<TaskListTile> createState() => _TaskListTileState();
 }
 
-
-
 class _TaskListTileState extends State<TaskListTile> {
 
-  bool? isChecked = false;
-
-  @override
-  void initState() {
-    super.initState();
-
-    isChecked = widget.task.isChecked;
-  }
-
   @override
   Widget build(BuildContext context) {
-    return ListTile(
-      leading: Transform.scale(
-        scale: 1.5,
-        child: Checkbox(
-            value: isChecked,
-            shape: const CircleBorder(),
-            fillColor:
-            MaterialStateColor.resolveWith((states) => Colors.transparent),
-            side: MaterialStateBorderSide.resolveWith(
-                    (states) => BorderSide(width: 1.5, color: widget.task.statusColor)),
-            checkColor: widget.task.statusColor,
-            onChanged: (bool? value) {
-              //  TODO API-Call here
-              setState(() {
-                isChecked = value;
-              });
-            }),
-      ),
-      trailing: Row(
-        mainAxisSize: MainAxisSize.min,
-        children: [
-          Padding(
-            padding: const EdgeInsets.all(paddingSmall),
-            child: StaticProgressIndicator(
-              progress: widget.task.progress,
-              color: widget.task.statusColor,
+    return Dismissible(
+      confirmDismiss: widget.confirmDismiss,
+        onDismissed: widget.onDismissed,
+        key: ValueKey(widget.index),
+        secondaryBackground: const ColoredBox(
+            color: Colors.red,
+            child: Align(
+              alignment: Alignment.centerRight,
+              child: Padding(
+                padding: EdgeInsets.all(paddingMedium),
+                child: Icon(Icons.delete),
+              ),
+            )
+        ),
+        background: const ColoredBox(
+          color: Colors.green,
+          child: Align(
+            alignment: Alignment.centerLeft,
+            child: Padding(
+              padding: EdgeInsets.all(paddingMedium),
+              child: Icon(Icons.done),
```

Comment:
change the color and icon when task is already in state done or remove the ability to reopen a task on left swipe